### PR TITLE
(fix) O3-3890 Help menu (and menu items) should use a standard Carbon…

### PR DESCRIPTION
**Pull request by Samstar10**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [x] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
I have updated the help menu to use standard carbon menu items. This PR is related to this PR https://github.com/openmrs/openmrs-esm-user-onboarding/pull/21

## Screenshots
Screenshot 2024-09-26 at 16 42 54

## Related Issue
https://openmrs.atlassian.net/browse/O3-3890

### CODE REVIEW

File: packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
       className={styles.helpButton}
-      href="https://o3-docs.openmrs.org"
-      rel="noopener noreferrer"
-      renderIcon={ArrowUpRight}
-      target="_blank"
-    >
-      {t('docs', 'Docs')}
-    </Link>
+      label={t('documentation', 'Documentation')}
+      onClick={() => window.open('https://o3-docs.openmrs.org', '_blank')}
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx:12]: Can we move the onClick handler to a function that's wrapped in a useCallback hook?

File: packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx
Comment (denniskigen) [inline packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx:1]: I wouldn’t expect these changes to be included here. Bad rebase?

File: packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
       className={styles.helpButton}
-      href="https://talk.openmrs.org"
-      rel="noopener noreferrer"
-      renderIcon={ArrowUpRight}
-      target="_blank"
-    >
-      {t('communityforum', 'Community forum')}
-    </Link>
+      label={t('communityforum', 'Community forum')}
+      onClick={() => window.open('https://talk.openmrs.org', '_blank')}
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx:12]: Can we wrap the onClick handlers in use callback hooks? Additionally, for security, it’s good to add “noreferrer noopener” to external links.

```jsx
const handleClick = useCallback(() => {
  window.open('https://talk.openmrs.org', '_blank', 'noopener,noreferrer');
}, [url]);

**Comment by denniskigen** [comment]
One challenge with this approach is that we can't leverage the `renderIcon` prop of the [MenuItem](https://react.carbondesignsystem.com/?path=/docs/components-menu--overview) component. To leverage it, the `MenuItem` must have a [Menu](https://react.carbondesignsystem.com/?path=/docs/components-menu--overview) component as its direct parent, and `Menu`'s `mode` prop should be set to `basic`.

File: packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx (hunk 1 above)
Comment (Samstar10) [reply packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx:12]: Noted

File: packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx
Comment (Samstar10) [reply packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx:1]: Yes, a mistake on my end, I'll fix it

File: packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx (hunk 2 above)
Comment (Samstar10) [reply packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx:12]: Noted

**Comment by brandones** [comment]
@Samstar10 Are you planning on continuing work on this? When? Is it clear to you how to proceed?

**Comment by Samstar10** [comment]
> @Samstar10 Are you planning on continuing work on this? When? Is it clear to you how to proceed?

Hello @brandones , yes, I have already completed it actually, just having some troubles with pushing it, hoping to be helped during today's coffee break.

**Comment by denniskigen** [comment]
Mind solving the conflicts, @Samstar10?

**Comment by Samstar10** [comment]
> Mind solving the conflicts, @Samstar10?

Yes, currently working on that.

File: packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
-import { ArrowUpRight } from '@carbon/react/icons';
+import { ArrowUpRight } from '@carbon/react/icons'
 import { Link } from '@carbon/react';
 
 const ContactUs = () => {
   const { t } = useTranslation();
   return (
     <Link
       className={styles.helpButton}
+      style={{ textDecoration: 'none' }}
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx:12]: This should move to `styles.scss`.

File: packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-import { ArrowUpRight } from '@carbon/react/icons';
+import { ArrowUpRight } from '@carbon/react/icons'
 import { Link } from '@carbon/react';
 
 const Docs = () => {
   const { t } = useTranslation();
   return (
     <Link
       className={styles.helpButton}
+      style={{ textDecoration: 'none' }}
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/docs.component.tsx:12]: Ditto.

File: packages/apps/esm-help-menu-app/src/help-menu/components/release-notes.component.tsx [hunk 5]
```diff
@@ -10,6 +10,7 @@ const ReleaseNotes = () => {
   return (
     <Link
       className={styles.helpButton}
+      style={{ textDecoration: 'none' }}
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/release-notes.component.tsx:13]: Ditto

File: packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss [hunk 6]
```diff
@@ -6,8 +6,13 @@
...
   justify-content: space-between;
+  padding: 0.5rem 0.5rem;
   color: black !important;
   &:focus {
     outline: none;
   }
 }
+
+.helpButton:hover {
+  background: #e8e8e8;
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss:17]:
```suggestion
  background: colors.$white-hover;
```

https://carbon-elements.netlify.app/colors/examples/preview/

File: packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss [hunk 7]
```diff
@@ -6,8 +6,13 @@
   display: flex;
   align-items: center;
   justify-content: space-between;
+  padding: 0.5rem 0.5rem;
```
Comment (denniskigen) [inline packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss:9]:
```suggestion
  padding: layout.$spacing-03;
```

nit: `padding: layout.spacing-03` is shorthand for `padding-top: 0.5rem; padding-right: 0.5rem; padding-bottom: 0.5rem; padding-left: 0.5rem`;

File: packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx
Comment (denniskigen) [inline packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx:1]: Why are we deleting this file? Seems unrelated to whatever else we're doing.

File: packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss (hunk 6 above)
Comment (Samstar10) [reply packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss:17]: @denniskigen this variable doesn't seem to exist

File: packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx (hunk 3 above)
Comment (Samstar10) [reply packages/apps/esm-help-menu-app/src/help-menu/components/contact-us.component.tsx:12]: @denniskigen for some reason whenever I move this to the styles.css it does not work. This is the only way it works.

File: packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx
Comment (Samstar10) [reply packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx:1]: I was trying to sort out the bad rebase. Working on fixing it now.

File: packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss (hunk 7 above)
Comment (Samstar10) [reply packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss:9]: Noted.

File: packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss (hunk 6 above)
Comment (denniskigen) [reply packages/apps/esm-help-menu-app/src/help-menu/components/styles.scss:17]: You can use `colors.$gray-10-hover` instead then:

CleanShot 2024-10-22 at 9  41 37@2x

**Comment by brandones** [comment]
Oh boy. Can we get this cleaned up and in? Thank you for your help @ibacher

**Comment by brandones** [comment]
Hello, @Samstar10 , are you still working on this?

**Comment by brandones** [comment]
Closing as stale.
